Redirect user to first step of daisy chain if not recognised

JamieDraper opened the pull request:
## Summary

https://wemove-europe.monday.com/boards/1864533892

Previously, unrecognised users who submitted the tweet form were not being directed to the next step in the daisy chain. This happened because ActionKit changes unrecognised users to recognised after the tweet form submission, but doesn’t automatically advance them to the following page, and instead reloads the existing tweet page in a 'user recognised' state.

This update ensures unrecognised users are immediately redirected to the first step of the daisy chain (the signup page), where we can collect their details before they reach the tweet form.

The redirect logic was added at the start of the prepareTweet() function, which runs once the ActionKit context is fully loaded. This ensures we can safely check actionkit.context.recognized_user before proceeding. If the user is unrecognised and daisy_chain_action is defined, we redirect them to the corresponding /sign/ page and stop further script execution.

## Testing
https://action.wemove.eu/call/2025-10-15-aaron-onboarding-tweet-EN
correctly redirects to
https://action.wemove.eu/sign/2025-10-15-aaron-onboarding-petition-EN/

https://action.wemove.eu/call/2025-10-15-aaron-onboarding-tweet-EN?akid=.2174112.NP8w5k&action_id=12613728&consent=inline
does not redirect (user is recognised)

JamieDraper committed:
Redirect user to first step of daisy chain if not recognised

aaronelliotross commented on WeMove/component-call_page-tweet.html:
```diff
@@ -16,6 +16,14 @@ <h2 class="text-white">
 
 <script>
+  const handleUnrecognizedUser = () => {
+    const daisyChainAction = {{ page.custom_fields.daisy_chain_action|json }};
```
Are you using the json filter to quote the value safely? Or copied from the twitter messages part? It is kind of a neat idea if it works, generates valid javascript no matter what the content is...

JamieDraper replied:
Yes, I was looking at how that json filter works and thought this way it would also handle escaping any special characters, though to be fair it's possibly not needed here since we have control over what the campaign value is...

I could change to `'{{ page.custom_fields.daisy_chain_action }}'` if you think that reads better!

aaronelliotross replied:
Nah, it's good - just making sure I understand. =)

aaronelliotross commented on WeMove/component-call_page-tweet.html:
```diff
@@ -16,6 +16,14 @@ <h2 class="text-white">
+    const daisyChainAction = {{ page.custom_fields.daisy_chain_action|json }};
+    const recognizedUser = actionkit.context?.recognized_user;
+    if (!recognizedUser && daisyChainAction) {
+      window.location.replace(`/sign/${daisyChainAction}/`);
```
One, smart to use replace to avoid the back button! Two, this should pass along the query string - e.g. if there are source codes we don't want to lose them. I'm 99% sure you need to manually pass them along, but not 100%, can you verify?

JamieDraper replied:
Ahh of course, you're right, I've made changes to pass them along

aaronelliotross commented on WeMove/component-call_page-tweet.html:
```diff
@@ -56,6 +64,10 @@ <h2 class="text-white">
   }
 
   const prepareTweet = () => {
+    if (handleUnrecognizedUser()) {
```
I think you need a return True at the end of the handleUnrecognizedUser() for this extra bit of safety to work.

What do you think about inlining the logic from the function? I think it might be a bit clearer. But either way!

JamieDraper replied:
Ah I think you're right, I don't think there's much to be gained currently from it being tucked into it's own function really, plus if it's inline we can just do the early return inside prepareTweet, I agree that it's clearer

aaronelliotross reviewed (changes requested):
Couple little questions ! And one request to mass along the query string when redirecting.

JamieDraper committed:
Tweet page redirect edits

- Move redirect logic inside prepareTweet
- Pass url params on when redirecting
